LDAR_Sim/src/methods/company.py

```diff
@@ -171,58 +173,123 @@ def flag_sites(self):
+        site_wl = self.site_watchlist
+        # Go through Each Candidate site, and add it to the watchlist
+        # A watchlist keeps track of sites with leaks that have not yet been flagged
+        for site in self.candidate_flags:
+            facility_ID = site['site']['facility_ID']
+            if facility_ID not in site_wl:
+                site_wl.update(
+                    {facility_ID: {
+                        'site': site['site'],
+                        'dates': [self.state['t'].current_date],
+                        'measured_emis_rates': [site['site_measured_rate']],
+                        'measured_vent_rates': [site['venting']],
+                        'effective_emis_rate': site['site_measured_rate'],
+                        'effective_vent_rate': site['venting']
+                    }})
+            else:
+                site_wl[facility_ID]['dates'].append(self.state['t'].current_date)
+                site_wl[facility_ID]['measured_emis_rates'].append(site['site_measured_rate'])
+                site_wl[facility_ID]['measured_vent_rates'].append(site['venting'])
+                # Calculate effective emissions and vent rates
+                site_wl[facility_ID]['effective_leak_rate'] = self.aggregate_by(
+                    site_wl[facility_ID]['measured_emis_rates'],
+                    self.config['follow_up']['redundancy_filter']
+                )
+                site_wl[facility_ID]['effective_vent_rate'] = self.aggregate_by(
+                    site_wl[facility_ID]['measured_vent_rates'],
+                    self.config['follow_up']['redundancy_filter']
+                )
+
+        # Sort Watchlist by emissions rate
+        site_wl = {k: v for k, v in sorted(
+            site_wl.items(),
+            key=lambda x: x[1]['effective_emis_rate'],
+            reverse=True)
+        }
+
+        # Get instant follow-up sites and init follow-up sites based on follow_up delay
+        IFU_sites = []
+        if self.config['follow_up']['instant_threshold']:
+            target_sites = []
+            for sdx, site in site_wl.items():
+                if site['effective_emis_rate'] > self.config['follow_up']['instant_threshold']:
+                    IFU_sites.append(site)
+                elif len(site['measured_emis_rates']) > self.config['follow_up']['delay']:
+                    target_sites.append(site)
+        else:
+            target_sites = [site for sdx, site in site_wl.items()
+                            if len(site['measured_emis_rates'])
+                            > self.config['follow_up']['delay']]
+
+        # Get portion of sites and threshold filtered sited
+        if self.config['follow_up']['interaction_priority'] == 'proportion':
+            n_sites_to_flag = len(target_sites) * self.config['follow_up']['proportion']
+            n_sites_to_flag = int(math.ceil(n_sites_to_flag))
+            target_sites = [site for site in target_sites[:n_sites_to_flag]
```

Comment:
Is 'target_sites' where sites that require follow-up go? Is it dangerous to keep redefining it? Could different companies change this before follow-up occurs?

Reply:
target sites is a local variable, it will not be effected by other companies. But they are sites that have passed the delay / grace period and have to be triaged by followup parameters. I have changed this, to a different object but it isnt typically dangerous to do inside of a list comprehension.